src/lib.rs

```diff
@@ -150,10 +190,17 @@
 #![warn(missing_docs)]
 #![cfg_attr(feature = "nightly", feature(panic_handler))]
 
+#![cfg_attr(all(not(feature = "use_std"), not(test)), no_std)]
```

Comment:
I mean putting the crate in the module it's needed, not up here. This is a core crate of the Rust ecosystem and is likely to be used as an example for downstream code (e.g. idioms and such), so I'd prefer to have it as idiomatic as possible.